### ISSUE TITLE
Infinite loop in `MultithreadConcurrentQueue` constructor

### ISSUE THREAD

**Issue by MichaelScofield** [conversant/disruptor#8]
When parameter `capacity` is `Integer.MAX_VALUE` in `MultithreadConcurrentQueue`'s constructor, 
```
    public MultithreadConcurrentQueue(final int capacity) {
        int c = 1;
        while(c < capacity) c <<=1;
        size = c;
        mask = size - 1L;
        buffer = (E[])new Object[size];
    }
```
it goes into infinite loop at line `while(c < capacity) c <<=1;`.

Because, `c` will left shift to `2^30`, which is < `Integer.MAX_VALUE`; and then `2^31` - unfortunately, it's `-2147483648` in signed integer in Java. And after that, `c` will be 0 forever, causing an infinite loop.

A simple fix would be: `while (c < capacity && c << 1 > 0) c <<= 1;`. However, it will break the javadoc in the constructor:
```
Note: actual capacity will be the next power of two larger than capacity.
```

**Comment by jac18281828** [comment]
Thanks for your problem report!   In my view its a bit _thick_ to set capacity to Integer.MAX_VALUE.   However, I can fix the problem by testing the argument, since Integer.MAX_VALUE will be a power of two I'll just set the capacity to this rather than modifying the interface.

**Comment by jac18281828** [comment]
Released in 1.2.12
